userguide/install.rst

```diff
@@ -1,35 +1,112 @@
 
 Start the installer and follow the on-screen instructions that will walk you through the installation process. That's it! Have fun exploring Ubuntu Touch!
 
-.. note::
+If you're an experienced android developer and want to help us bring Ubuntu Touch to more devices, visit the :doc:`porting section </porting/introduction>`.
 
-    If your device is not detected automatically when using the snap installer, then unplug your device, open a terminal and enter the following command:
+Troubleshooting
+---------------
 
-    .. code-block:: text
+If you need help, you can join UBports' support channels on `telegram <https://t.me/WelcomePlus>`_ or `matrix <https://matrix.to/#/!KwdniMNeTmClpgHkND:matrix.org?via=matrix.org&via=ubports.chat&via=disroot.org>`_ or ask a question `in the forum <https://forums.ubports.com/>` or on `askubuntu <https://askubuntu.com>`. If you believe that the UBports Installer is not working correctly, you can also `file a new issue <https://github.com/ubports/ubports-installer/issues/new>` to help us solve the problem.
 
-        sudo snap connect ubports-installer:raw-usb
+Connection lost during installation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-    Then restart the installer, plug in your device and give it another try.
+Try a different USB cable and a different USB port on your computer. Cheap and old cables tend to lose connection during the installation.
 
+Windows Defender prevents Installer from starting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-If you're an experienced android developer and want to help us bring Ubuntu Touch to more devices, visit the :doc:`porting section </porting/introduction>`.
+We have contacted Microsoft about this problem, but they seem to think it's `enjoyable <https://twitter.com/Windows/status/1014984163433295875>`__. To dismiss the warning, click on "More Information", then select "Run anyway".
+
+Device not detected
+^^^^^^^^^^^^^^^^^^^
+
+Custom platform tools
+"""""""""""""""""""""
+
+You might want to try using custom adb and fastboot binaries. For that, tick the "Custom tools" checkbox in the options and specify the path to adb and fastboot in the text-boxes below.
+
+Virtual Machine
+"""""""""""""""
+
+Make sure the virtual machine is allowed to access the USB port.
+
+Windows ADB drivers
+"""""""""""""""""""
+
+Install the `universal adb driver <http://adb.clockworkmod.com/>`_ and reboot your computer.
+
+Fairphone 2 Vendor-ID
+"""""""""""""""""""""
+
+Run ``echo 0x2ae5 >> ~/.android/adb_usb.ini`` in the terminal and restart the installer. This does not affect the snap package.
+
+Missing udev-rules
+""""""""""""""""""
+
+If the device not detected on Linux and packages other than snap, you might be missing **udev-rules**.
+
+1. See if ``/etc/udev/rules.d/51-android.rules`` exists and contains the rules below. If not, add them to the file and run
+   ``sudo service udev restart`` or
+   ``sudo udevadm control --reload-rules && udevadm trigger``.
```

Comment:
These do not appear on separate lines. Also, is `service udev restart` appropriate with systemd?

Reply:
That's why the second command is there